### PR TITLE
Connect Qualitative Analysis UI to API

### LINKED ISSUE #4341: Analysis questions: Questions list store
A data store that will be handling API calls and the questions and responses to them.

### LINKED ISSUE #4349: Display analysis questions columns: Data Table update
displayed to the right of the last transcript/translation for the particular audio question

### LINKED ISSUE #4350: Display analysis questions columns: Single Submission Modal update
Displayed similarly to #4349

### LINKED ISSUE #4351: Display analysis questions columns: Data Downloads exported columns selection
Include new columns in "Select questions to be exported". Make sure to use the same utility functions in all places that display qualitative questions columns

### LINKED ISSUE #4361: Allow analysis questions in the asset endpoint's advanced_features
keyword: qualitative

### PR DESCRIPTION
**Pull request by magicznyleszek**
## Checklist

1. [x] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [ ] Write a description of your work suitable for publishing on [our forum](https://community.kobotoolbox.org/tag/release-notes)
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Make adding Qualitative Analysis Question work in the UI. Also make the question responses appear in Data Table and in the Data Downloads.

## Code Review Notes

Front end:
- removed bunch of leftover functions (and their tests) that were (already) no longer being used:
  - `getSupplementalTranscriptPath`
  - `getSupplementalTranslationPath`
  - `getSupplementalDetailsPaths`
  - `getRowSupplementalResponses`
- moved the `AnalysisTabContent` into `analysis` directory for clarity (as it is setting up all the analysis questions code)
- `analysisQuestions.actions` and `analysisQuestions.reducer` are being updated to work with API calls
- a clanup and improvement of types of many things touched during work
- hide `qual_auto_keyword_count` (Keyword Search) question from UI (leaving code for upcoming work on this)
- `TransxSelector` now has `size` and `type` options
- added few tests

Back end:
- @jnm knows

## Related issues

Includes Back-end work from `feature-analysis-module` branch by @dorey and some more coding by @jnm 

Closes #4341
Closes #4361
Closes #4351
Closes #4350
Closes #4349

### CODE REVIEW

**Review by srartese — COMMENTED** [review]
I followed the changes you made and it make sense to me. 
I'm not the best for python review so we should add a second set of eyes there. There is also just a lot of changes so it would be helpful. 

One bug I found was when testing locally I'm not able to open forms that have audio questions. I tried creating a new form, editing an old one and neither allows me to open when it has the audio question. 
Screen Shot 2023-09-06 at 3 35 55 PM

**Comment by magicznyleszek** [comment]
@srartese Have you tried rebuilding kpi and kobocat? And if this wouldn't help, running migrations? And it would be helpful if you'd give some more details on that error - try console or container logs

**Comment by srartese** [comment]
I've tested it and looked at the analysis feature compared to the figma, it looks great and works smoothly. There are a few more things included locally than what appears in figma and the icon colors are off but it doesn't seem like this is the right PR for that kind of feedback.

**Review by srartese — APPROVED** [review]

**Comment by srartese** [comment]
Can't seem to remove myself after having approved some of the changes earlier.  Please ignore me for the time being

File: jsapp/js/components/processing/analysis/responseForms/selectOneResponseForm.component.tsx
Comment (LMNTL) [inline jsapp/js/components/processing/analysis/responseForms/selectOneResponseForm.component.tsx:21]: If I delete a single choice/radio option, it still shows up, even after exiting and re-entering the Analysis tab:

Before:
After:

It looks like the Select Multiple/checkbox fields hide deleted options in the Analysis tab, though.

**Review by LMNTL — CHANGES_REQUESTED** [review]
Looking good so far! I found some things that may need fixing, let me know if any of them are known issues or should be referred to the back-end team.

One general comment - the UI for Qualitative Analysis starts to break around 500-600px width, and some fields become hard to use/see as a consequence:

image

It's not part of this scope and wasn't introduced in this PR, so it's not blocking this review. But Quantitative Analysis does seem like the sort of feature that people in the field would end up using on lower-end mobile devices.

**Comment by magicznyleszek** [comment]
> Looking good so far! I found some things that may need fixing, let me know if any of them are known issues or should be referred to the back-end team.
> 
> One general comment - the UI for Qualitative Analysis starts to break around 500-600px width, and some fields become hard to use/see as a consequence:
> 
> image
> 
> It's not part of this scope and wasn't introduced in this PR, so it's not blocking this review. But Quantitative Analysis does seem like the sort of feature that people in the field would end up using on lower-end mobile devices.

Made a Notion page: https://www.notion.so/kobotoolbox/Improve-Qualitative-Analysis-UI-for-small-screens-7d22a2ea9d4a4479b9d4744f7d6c5d09?pvs=4

File: jsapp/js/components/processing/analysis/analysisQuestions.reducer.ts
Comment (magicznyleszek) [reply jsapp/js/components/processing/analysis/analysisQuestions.reducer.ts:128]: I tried reproducing this, but am unable to. I did merge some two other PRs first, so maybe they had a fix in them :shrug:.

I was considering adding these lines in `activateTab` function from `singleProcessingStore.ts`:

```ts
    // When we leave Analysis tab, we need to reset the flag keeping up status
    // of unsaved changes. This way it's not blocking navigation after leaving
    // the tab while having unsaved changes.
    this.setAnalysisTabHasUnsavedChanges(false);
```

File: jsapp/js/components/processing/analysis/responseForms/integerResponseForm.component.tsx
Comment (magicznyleszek) [reply jsapp/js/components/processing/analysis/responseForms/integerResponseForm.component.tsx:80]: I pushed a fix for FE to allow sending empty. @jnm will fix BE to allow receiving empty value

File: jsapp/js/components/processing/analysis/responseForms/selectOneResponseForm.component.tsx
Comment (magicznyleszek) [reply jsapp/js/components/processing/analysis/responseForms/selectOneResponseForm.component.tsx:22]: I forgot to filter out choices with `options.deleted` flag for `qual_select_one`…

File: jsapp/js/components/processing/analysis/analysisQuestions.reducer.ts
Comment (magicznyleszek) [reply jsapp/js/components/processing/analysis/analysisQuestions.reducer.ts:128]: > […] I did merge some two other PRs first, so maybe they had a fix in them 🤷.

I checked and neither #4761 nor #4751 have any changes that would seem like a fix for that

File: jsapp/js/components/processing/analysis/responseForms/integerResponseForm.component.tsx
Comment (LMNTL) [reply jsapp/js/components/processing/analysis/responseForms/integerResponseForm.component.tsx:80]: Looking good, once the backend fix for this is merged it has my approval.

**Review by LMNTL — APPROVED** [review]

**Pull request merged**
